compiler/rustc_infer/src/infer/freshen.rs

```diff
@@ -253,3 +193,52 @@ impl<'a, 'tcx> TypeFolder<'tcx> for TypeFreshener<'a, 'tcx> {
         }
     }
 }
+
+impl<'a, 'tcx> TypeFreshener<'a, 'tcx> {
+    fn fold_infer_ty(&mut self, v: ty::InferTy) -> Option<Ty<'tcx>> {
```

Comment:
Please document that this was outlined on purpose so no one comes around in the future and is like "oh it has only one call site, let's inline it"